cmd/producer/main.go

```diff
@@ -38,21 +35,27 @@ func main() {
 	log.Event(ctx, "loaded config", log.INFO, log.Data{"cfg": cfg})
 
 	pChannels := kafka.CreateProducerChannels()
+	pChannels.LogErrors(ctx, "Producer error")
 	pConfig := &kafka.ProducerConfig{
 		KafkaVersion: &cfg.KafkaVersion,
 	}
 
-	producer, err := kafka.NewProducer(ctx, brokers, cfg.ConsumerTopic, pChannels, pConfig)
+	producer, err := kafka.NewProducer(ctx, cfg.KafkaAddr, cfg.ConsumerTopic, pChannels, pConfig)
 	if err != nil {
 		log.Event(ctx, "failed to create kafka producer", log.FATAL, log.Error(err))
 		os.Exit(1)
 	}
 
+	// wait for the producer to be ready before attempting to send the message
+	<-pChannels.Ready
```

Comment:
I think it might be sensible to have a `select`  here with an option for `time.After(x)` with some sensible default. If the default expires we can exiting with a warning message. As it stands this can wait indefinitely